github/git_refs.go

```diff
@@ -56,15 +59,75 @@ func (s *GitService) GetRef(ctx context.Context, owner string, repo string, ref
 		return nil, nil, err
 	}
 
-	r := new(Reference)
-	resp, err := s.client.Do(ctx, req, r)
+	buf := new(bytes.Buffer)
+	resp, err := s.client.Do(ctx, req, buf)
 	if err != nil {
 		return nil, resp, err
 	}
 
+	rbuf := bytes.NewBuffer(buf.Bytes())
+	r := new(Reference)
+
+	err = json.NewDecoder(rbuf).Decode(r)
+	if err != nil {
+		_, ok := err.(*json.UnmarshalTypeError)
+		if ok {
+			// multiple responses
+			return nil, resp, nil
+		} else {
+			return nil, resp, err
+		}
+	}
```

Comment:
I think this can be cleaner if we reduce the indentation by one level:

```Go
err = json.NewDecoder(rbuf).Decode(r)
if _, ok := err.(*json.UnmarshalTypeError); ok {
	// Multiple refs, means there wasn't an exact match.
	return nil, resp, nil
} else if err != nil {
	return nil, resp, err
}
```

Similar can be applied below.